### ISSUE TITLE
Server ignores interact packet action type

### ISSUE THREAD

**Issue by thebigsmileXD** [PocketMine/PocketMine-MP#3684]
As you may or may not know, when you interact with some entities like boats, minecarts or wolves you see a little button on singleplayer. PocketMine doesn't use them in any way (you see them, but.. Yeaa). Could an interact button packet be added? When you play in local multiplayer it works.

**Comment by SkygridRex** [comment]
not if you knew that the aforementioned characteristics are missing or poorly programmed

**Comment by AndreyNazarchuk** [comment]
@SkygridRex That's why he's asking for them to be implemented...
:+1:

**Comment by SkygridRex** [comment]
@thebigsmileXD This is not a Forum! please visit forums.pocketmine.net
(Oh No! I'm Converting in @PEMapModder xD)

**Comment by AndreyNazarchuk** [comment]
@SkygridRex or you are just saying that with no idea why you are saying it. @thebigsmileXD made a request for enhancement. That is one of the correct things to ask here. By saying this I am turning it into a forum, I just had to point out how naive you were being.

TL;DR
Don't speak if you don't know what you are talking about.

**Comment by thebigsmileXD** [comment]
Thanks alot. :). If the enchantment would be done i could continue coding Wolf AI
I currently hit them instead of letting them sit

**Comment by PEMapModder** [comment]
@SkygridRex he is not using this as a forum. He pointed out a possible improvement in the API.

**Comment by SkygridRex** [comment]
@AndreyNazarchuk You can guess what I do? Mirame press the unsubscribe button. You say that is not what I'm talking about? I am not foolish enough to see that for example the code tameable.php is insufficient. in short, everything is for lack of code.
@thebigsmileXD Visit this page where you can see an example of what you want in operation: https://github.com/onebone/Boat
@PEMapModder not waste time with you!
If you think this comment disrespectful my sincere apologies

**Comment by thebigsmileXD** [comment]
@SkygridRex This is still not implemented to pocketmine. And thats on Tap. But e.g Wolves get hurt on tap, but on hold / button press they are feeded. Want the wolves to run at you and kill you? And it is disrespectful. Also this doesn't only effect Tameable.php. You gave the right example, boats. They are not tameable. They work instantly for everyone

**Comment by SkygridRex** [comment]
@thebigsmileXD In what language I say that is not implemented due to lack of code?

**Comment by PEMapModder** [comment]
The problem is that PocketMine ignores the [action ID](https://github.com/NiclasOlofsson/MiNET/blob/master/src/MiNET/MiNET/Net/MCPE%20Protocol.xml#L345) of the interact packet, so clicks and long clicks are considered as the same action of hitting.

**Comment by SkygridRex** [comment]
so, rewrite InteractPacket.php

**Comment by thebigsmileXD** [comment]
Just saying, thats why i ask TO ADD THEM

**Comment by SkygridRex** [comment]
@thebigsmileXD in multiplayer local not work those buttons

**Comment by thebigsmileXD** [comment]
https://bugs.mojang.com/browse/MCPE-11991

**Comment by PEMapModder** [comment]
This is actually possible with DataPacketReceiveEvent.
Nevertheless, it seems that PocketMine just forgot to check the action.
